packages/expo-image-picker/ios/ImagePickerModule.swift

```diff
@@ -0,0 +1,169 @@
+    }
+
+    guard permissions.hasGrantedPermission(usingRequesterClass: CameraPermissionRequester.self) else {
+      return promise.reject(MissingCameraPermissionError())
+    }
+
+    let pickingContext = PickingContext(promise: promise,
+                                        options: options,
+                                        imagePickerDelegate: ImagePickerDelegate(onMediaPickingResultHandler: self, hideStatusBarWhenPresented: options.allowsEditing))
+    self.launchImagePicker(sourceType: .camera, pickingContext: pickingContext)
+  }
+
+  private func launchImageLibrary(options: PickingOptions, promise: Promise) {
+    guard let _ = self.appContext?.permissions else { return promise.reject(PermissionsModuleNotFoundError()) }
+
+    let pickingContext = PickingContext(promise: promise,
+                                        options: options,
+                                        imagePickerDelegate: ImagePickerDelegate(onMediaPickingResultHandler: self, hideStatusBarWhenPresented: options.allowsEditing))
+    self.launchImagePicker(sourceType: .photoLibrary, pickingContext: pickingContext)
+  }
+
+
+  private func launchImagePicker(sourceType: UIImagePickerController.SourceType, pickingContext context: PickingContext) {
+    let options = context.options
+    let promise = context.promise
+    let picker = UIImagePickerController()
+
+    if (sourceType == .camera) {
+#if targetEnvironment(simulator)
+      return context.promise.reject(CameraUnavailableOnSimulatorError())
+#else
+      picker.sourceType = .camera
+      picker.cameraDevice = context.options.cameraType == .front ? .front : .rear
+#endif
+    }
+
+    if (sourceType == .photoLibrary) {
+      picker.sourceType = .photoLibrary
+    }
+
+    picker.mediaTypes = options.mediaTypes.toArray()
+    picker.videoExportPreset = options.videoExportPreset.toAVAssetExportPreset()
+    picker.videoQuality = options.videoQuality.toQualityType()
+    picker.videoMaximumDuration = options.videoMaxDuration
+
+    if (options.allowsEditing) {
+      picker.allowsEditing = options.allowsEditing
+      if (options.videoMaxDuration > 600) {
+        return promise.reject(MaxDurationWhileEditingExceededError())
+      }
+      if (options.videoMaxDuration == 0) {
+        picker.videoMaximumDuration = 600.0
+      }
+    }
+
+    picker.modalPresentationStyle = options.presentationStyle.toPresentationStyle()
+    picker.delegate = context.imagePickerDelegate
+    picker.presentationController?.delegate = context.imagePickerDelegate
+
+    guard let currentViewController = self.appContext?.utilities?.currentViewController() else {
+      return promise.reject(MissingCurrentViewController())
+    }
+
+    // Stora picking context as we're navigating to the different view controller (starting asynchrounus flow)
+    self.currentPickingContext = context
+    currentViewController.present(picker, animated: true, completion: nil)
+  }
+
+  // MARK: OnMediaPickingResultHandler
```

Comment:
I think it looks a bit nicer in the editor if you do 👇 — it renders a separator between the following sections

```suggestion
  // MARK: - OnMediaPickingResultHandler
```